test/builtin_dataset_mocks.py

```diff
@@ -1431,3 +1431,48 @@ def stanford_cars(info, root, config):
         make_tar(root, "car_devkit.tgz", devkit, compression="gz")
 
     return num_samples
+
+
+@register_mock
+def sun397(info, root, config):
+    images_root = root / "SUN397"
+    images_root.mkdir()
+
+    categories = ["abbey", "airplane_cabin", "wrestling_ring/indoor"]
+    category_keys = [f"/{category[0]}/{category}" for category in categories]
+
+    keys = []
+    for key in category_keys:
+        parts = key.split("/")
+        image_files = create_image_folder(
+            root=images_root.joinpath(*parts[1:-1]),
+            name=parts[-1],
+            file_name_fn=lambda idx: f"sun_{idx:05d}.jpg",
+            num_examples=5,
+        )
+
+        keys.extend([f"/{image_file.relative_to(images_root).as_posix()}" for image_file in image_files])
```

Comment:
After the image generation, we no longer care for the actual image path, but only its key.